### PR TITLE
[7.1] Remove Foshttpcache code, leaving that to ezplatform-http-cache

### PR DESCRIPTION
**Pull request by andrerom**
>  Part of https://jira.ez.no/browse/EZP-28065 Evaluate upgrading to fos-http-cache 2.x

Replaces #2013, similar to 7.0 deprecation removal in #2190, this removes all previously deprecated FOS http cache related code so that we can:
- Handle HttpCache purely in ezplatform-http-cache
    - Only a few things depending on FOSHttpCache remains here, code in ca7e4ef and twig usage in #2263, in both cases they can also work with 2.x so requirement is here relaxed to allow 2.x..
    - So we in turn can safely update to FOS HttpCache 2.x once that is ready for us at a later stage without having to consider having to bump major kernel version.
- One of several steps needed in order to start testing against Symfony 4.x

TODO:
- [x] Fix functional tests
- [x] Remove new changes to deprecated http cache impl in 9b45ac9
- [x] Setup instance with this and run functional test locally to figure out remaining cryptic errors, and..
- [x] Lookup the added `@todo`'s in yml file
 - [x] fos usage in config, like:
    ```yml
    ezpublish.siteaccess_match_listener:
        class: "%ezpublish.siteaccess_match_listener.class%"
        arguments: ["@ezpublish.siteaccess_router", "@event_dispatcher", "@fos_http_cache.user_context.request_matcher"]
        tags:
            - { name: kernel.event_subscriber }
    ```
- [x] Upgrade doc on hash generation to refer to how to use this using FOSHttpCache: https://github.com/ezsystems/ezpublish-kernel/blob/89ec5c6f30683058d8d479be557d72b84f485c6f/doc/specifications/cache/http/context_aware_http_cache.md#user-hash-generation

### CODE REVIEW

**Comment by andrerom** [comment]
@bdunogier Do I perhaps need to move over HashGenerator & User/Identity also?

**Comment by andrerom** [comment]
@vidarl For now `ezplatform-http-cache` works on both 6.13 and 7.0, so the kernel removal parts needs to stay, and continue to only remove if there.

Once we at later point branches off to use FOSHttpCache 2.2 or something, that will be exclusive for new 2.x releases, and bump major version of `ezplatform-http-cache`, at that point we can remove those passes.

File: eZ/Bundle/EzPublishCoreBundle/Tests/Cache/Warmer/ConfigResolverCleanupTest.php [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
- * @license For full copyright and license information view LICENSE file distributed with this source code.
- */
-namespace eZ\Bundle\EzPublishCoreBundle\Tests\Cache\Warmer;
-
-use eZ\Bundle\EzPublishCoreBundle\Cache\Warmer\ConfigResolverCleanup;
-use PHPUnit\Framework\TestCase;
-use stdClass;
-use Symfony\Component\DependencyInjection\Container;
-
-class ConfigResolverCleanupTest extends TestCase
```
Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/Tests/Cache/Warmer/ConfigResolverCleanupTest.php]: this should not be removed afaik, to be checked

File: eZ/Publish/Core/MVC/Symfony/Cache/Http/SymfonyCache/UserContextSubscriber.php [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-
-use FOS\HttpCache\SymfonyCache\UserContextSubscriber as BaseUserContextSubscriber;
-use Symfony\Component\HttpFoundation\Request;
-
-/**
- * Extends UserContextSubscriber from FOSHttpCache to include original request.
- *
- * {@inheritdoc}
- */
-class UserContextSubscriber extends BaseUserContextSubscriber
```
Comment (andrerom) [inline eZ/Publish/Core/MVC/Symfony/Cache/Http/SymfonyCache/UserContextSubscriber.php]: don't seem to be used anywhere, however if it is we might need to move this to ezplatform-http-cache

Comment (andrerom) [reply eZ/Publish/Core/MVC/Symfony/Cache/Http/SymfonyCache/UserContextSubscriber.php]: https://github.com/ezsystems/ezplatform-http-cache/pull/46

**Comment by andrerom** [comment]
🎉  Passing 🎆 

@bdunogier While there are lots of things to go over here and discuss, I especially kept one change in separate commit here, the change to SiteAccessMatchListener. As I can't fully put my finger on it but thinking there might alternative approaches to this one to avoid having to refer to fos service in kernel.

**Comment by andrerom** [comment]
Updated, open question above is no longer relevant, with changes in #2263 it's ok to keep depending on FOSHttpCache _(for using it's api's, as opposed to extending it which is purely left to external bundle)_ as long as we allow 2.x as well.

**Comment by andrerom** [comment]
Should be fixed now then: https://github.com/ezsystems/ezplatform/commit/11d438ad49a13b9cca885e61b29baad0344a291e

(restarting)

**Comment by lserwatka** [comment]
@alongosz we need to address this for 2.1

**Comment by andrerom** [comment]
@alongosz Kernel does not depend on `ezplatform-http-cache` so if you are afraid customers will hit this on upgrade only thing we can do is setup conflict with 0.4. Besides that this is now ready for review.

**Comment by alongosz** [comment]
> Kernel does not depend on ezplatform-http-cache so if you are afraid customers will hit this on upgrade only thing we can do is setup conflict with 0.4. Besides that this is now ready for review.

No, the change https://github.com/ezsystems/ezplatform/commit/11d438ad49a13b9cca885e61b29baad0344a291e should be enough.

**Pull request merged**